src/algorithms/WState.cpp

```diff
@@ -0,0 +1,30 @@
+#include "algorithms/WState.hpp"
+
+namespace qc {
+void fGate(QuantumComputation& qc, const Qubit i, const Qubit j, const Qubit k,
+           const Qubit n) {
+  const auto theta = std::acos(std::sqrt(1.0 / static_cast<double>(k - n + 1)));
+  qc.ry(j, -theta);
+  qc.z(j, qc::Control{i});
+  qc.ry(j, theta);
+}
+
+WState::WState(const Qubit nq) : QuantumComputation(nq) {
+  if (nq == 0) {
+    return;
+  }
+
+  name = "wstate_" + std::to_string(nq);
+  const auto top = nq - 1;
+
+  x(top);
```

Comment:
```suggestion
  x(nq- 1);
```
Eliminates two more lines 🙂